mlrun/feature_store/steps.py

```diff
@@ -181,6 +184,10 @@ def do(self, event):
             encoded_values.update(self._encode(feature, val))
         return encoded_values
 
+    def _prepare_category(self, f_name: str):
+        """replace (" " and "-") -> "_" """
+        return "_".join(("_".join(f_name.split())).split("-"))
```

Comment:
In my opinion this is much more readable:
```
return f_name.replace(' ', '_').replace('-', '_')
```

Also, if it's used only once, you may do it inline instead of a separate function.